internal/shellgen/scripts.go

```diff
@@ -25,6 +25,7 @@ type devboxer interface {
 	Lockfile() *lock.File
 	AllInstallablePackages() ([]*devpkg.Package, error)
 	InstallablePackages() []*devpkg.Package
+	IsUserShellFish() (bool, error)
```

Comment:
Not really introduced in this PR, but I think we should cut down on this interface pattern for pure data like this. (I admin to be much to blame for this)

There are some cases where the circular dependency is hard to avoid (for example if you need to call a function with side-effects that would create a cycle). If what you are passing in is pure data, cheap to compute (like in this case), then using a struct argument keeps code simpler and more linear.

Reply:
Yeah, I agree 💯 

In this particular package, there are more public APIs than I think are good. For example, `WriteScriptToFile` should maybe not be a public method. Having a tighter public interface, would enable a constructor which can take some of these arguments (as you suggest), minimizing the need for the interface.